cmd/run_scenario.go

```diff
@@ -33,23 +34,27 @@ func runScenarioCmd() *cobra.Command {
 }
 
 type scenarioRunner struct {
-	logger         *zap.SugaredLogger
-	scenario       string
-	runID          string
-	iterations     int
-	duration       time.Duration
-	waitForServer  time.Duration
-	clientOptions  cmdoptions.ClientOptions
-	metricsOptions cmdoptions.MetricsOptions
-	loggingOptions cmdoptions.LoggingOptions
+	logger          *zap.SugaredLogger
+	scenario        string
+	runID           string
+	iterations      int
+	duration        time.Duration
+	maxConcurrent   int
+	scenarioOptions []string
+	connectTimeout  time.Duration
+	clientOptions   cmdoptions.ClientOptions
+	metricsOptions  cmdoptions.MetricsOptions
+	loggingOptions  cmdoptions.LoggingOptions
 }
 
 func (r *scenarioRunner) addCLIFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&r.scenario, "scenario", "", "Scenario name to run")
 	fs.StringVar(&r.runID, "run-id", "", "Run ID for this run")
-	fs.IntVar(&r.iterations, "iterations", 0, "Iterations for the scenario (cannot be provided with duration)")
-	fs.DurationVar(&r.duration, "duration", 0, "Duration for the scenario (cannot be provided with iteration)")
-	fs.DurationVar(&r.waitForServer, "wait-for-server", 0, "Duration to try to connect to server before failing")
+	fs.IntVar(&r.iterations, "iterations", 0, "Override default iterations for the scenario (cannot be provided with duration)")
+	fs.DurationVar(&r.duration, "duration", 0, "Override duration for the scenario (cannot be provided with iteration)")
+	fs.IntVar(&r.maxConcurrent, "max-concurrent", 0, "Override max-concurrent for the scenario")
```

Comment:
If we really want to make the executor generic, these should be supplied via `--option`, I never got around to restructuring this.

Reply:
Can wait for when there's an executor that doesn't take these params.